app/src/main/java/com/nextcloud/talk/ui/dialog/MessageActionsDialog.kt

```diff
@@ -302,88 +306,70 @@ class MessageActionsDialog(
-                    }
-                }
-
-                override fun onError(e: Throwable) {
-                    Log.e(TAG, "error while sending reaction: $emoji")
-                }
-
-                override fun onComplete() {
-                    dismiss()
-                }
-            })
+    inner class ReactionAddedObserver : Observer<ReactionAddedModel> {
+        override fun onSubscribe(d: Disposable) {
+            // unused atm
+        }
+
+        override fun onNext(reactionAddedModel: ReactionAddedModel) {
+            if (reactionAddedModel.success) {
+                chatController.updateUiToAddReaction(
+                    reactionAddedModel.chatMessage,
+                    reactionAddedModel.emoji
+                )
+            }
+        }
+
+        override fun onError(e: Throwable) {
+            Log.e(TAG, "failure in ReactionAddedObserver", e)
+        }
+
+        override fun onComplete() {
+            dismiss()
+        }
     }
 
-    private fun deleteReaction(message: ChatMessage, emoji: String) {
-        val credentials = ApiUtils.getCredentials(user?.username, user?.token)
-
-        ncApi.deleteReaction(
-            credentials,
-            ApiUtils.getUrlForMessageReaction(
-                user?.baseUrl,
-                currentConversation!!.token,
-                message.id
-            ),
-            emoji
-        )
-            ?.subscribeOn(Schedulers.io())
-            ?.observeOn(AndroidSchedulers.mainThread())
-            ?.subscribe(object : Observer<GenericOverall> {
-                override fun onSubscribe(d: Disposable) {
-                    // unused atm
-                }
-
-                override fun onNext(genericOverall: GenericOverall) {
-                    Log.d(TAG, "deleted reaction: $emoji")
-                }
-
-                override fun onError(e: Throwable) {
-                    Log.e(TAG, "error while deleting reaction: $emoji")
-                }
-
-                override fun onComplete() {
-                    dismiss()
-                }
-            })
+    inner class ReactionDeletedObserver : Observer<ReactionDeletedModel> {
+        override fun onSubscribe(d: Disposable) {
+            // unused atm
+        }
+
+        override fun onNext(reactionDeletedModel: ReactionDeletedModel) {
+            if (reactionDeletedModel.success) {
+                chatController.updateUiToDeleteReaction(
+                    reactionDeletedModel.chatMessage,
+                    reactionDeletedModel.emoji
+                )
+            }
+        }
+
+        override fun onError(e: Throwable) {
+            Log.e(TAG, "failure in ReactionDeletedObserver", e)
+        }
+
+        override fun onComplete() {
+            dismiss()
+        }
     }
 
     companion object {
-        private const val TAG = "MessageActionsDialog"
+        private val TAG = MessageActionsDialog::class.java.simpleName
```

Comment:
why not a `const` anymore?

Reply:
it can't be a `const` when using `::class.java.simpleName`